service/history/workflow/workflow_task_state_machine.go

```diff
@@ -554,6 +572,8 @@ func (m *workflowTaskStateMachine) AddWorkflowTaskFailedEvent(
 			workflowTask.RequestID,
 			identity,
 			timestamp.TimeValue(workflowTask.StartedTime),
+			workflowTask.SuggestContinueAsNew,
+			workflowTask.HistorySizeBytes,
```

Comment:
I was a little unsure about these. I think probably it should recompute them here? (It seems logical that it should recompute every time `hBuilder.AddWorkflowTaskStartedEvent` is called, and not any other time.) But I'm not sure how speculative works...

Reply:
this is a failure, so it doesn't matter